### PR TITLE
Nomenclature surrounding how copper oxidizes in Minecraft

### PR DESCRIPTION
**Pull request by Jkmcameron**
This mapping differs from the one the others settled on because I don't agree with mapping this Interface to "Oxidizable" since there's no reason why the very same interface couldn't be used to create a plant that degrades over time. The name "Oxidizable" implies that it's usage is limited to metals, and "Corrodable" has the same issue, it is only "Degradable" that perfectly illustrates what this interface is responsible for without defining an implicit scope

### CODE REVIEW

**Review by haykam821 — CHANGES_REQUESTED** [review]
I don't think that this nomenclature is any better than what was suggested in #1830. Degradation could imply that part of the block is broken; meanwhile, oxidization is the only behavior this interface is used for in vanilla.

**Comment by OroArmor** [comment]
However, the features that the name `Degradation` implies is wider than `Oxidizable`. As well, (I haven't checked the code), copper still `oxidizes` even though it could be surrounded by blocks on all sides

**Comment by i509VCB** [comment]
The name of this interface should relate to current uses imo. My guess is that only copper like blocks use the interface.

**Comment by OroArmor** [comment]
Potentially, but for modders it shows a different feature aspect, something that these blocks can evolve over time. I do agree that things should be how they are in vanilla, but the use of this might make it a better choice to make it have a more abstract name.

**Comment by Jkmcameron** [comment]
> The name of this interface should relate to current uses imo. My guess is that only copper like blocks use the interface.

> Degradation could imply that part of the block is broken; meanwhile, oxidization is the only behavior this interface is used for in vanilla.

What logic is this? Are we creating the mappings for Mojang or for the modding community? The name shouldn't be limited by its vanilla usage, it should be limited by its full range of potential usages. Sure, Mojang added this bit of code solely for the purpose of oxidizing copper, but a modder could just as easily use this very same code to create a plant that wilts over time.

**Comment by jaskarth** [comment]
`Oxidizable` is the best choice here because only copper uses it currently. Instead, you should add javadoc to the class telling modders that it can have wider uses.

File: mappings/net/minecraft/block/CopperBlock.mapping [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/class_5548 net/minecraft/block/Copper_Block
```
Comment (Gegy) [inline mappings/net/minecraft/block/CopperBlock.mapping]:
```suggestion
CLASS net/minecraft/class_5548 net/minecraft/block/CopperBlock
```

**Comment by Jkmcameron** [comment]
> `Oxidizable` is the best choice here because only copper uses it currently. Instead, you should add javadoc to the class telling modders that it can have wider uses.

Following that logic we should name it "CopperOxidizable", wouldn't want anyone getting the wrong idea and thinking this interface was meant to be used by anything other than copper.

**Comment by jaskarth** [comment]
Well, you can use the interface for anything that you want to, really. The names of classes should be used to describe the behavior by vanilla, as modders can use it for a wide range of behaviors. Likewise, we don't need to indicate to modders how to use a class as it's ultimately up to them.

**Comment by YanisBft** [comment]
Closing this as people seem to be against this. Please open an issue if you still think your proposal is better

**Pull request closed without merging**